x/stakeibc/keeper/hooks.go

```diff
@@ -53,20 +54,56 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 				}
 			}
 		}
+
+		// DELEGATE FROM DELEGATION ACCOUNT
 		delegateInterval := int64(k.GetParam(ctx, types.KeyDelegateInterval))
 		if epochNumber%delegateInterval == 0 {
+			// get Gaia LC height
 			k.ProcessDelegationStaking(ctx)
 		}
+
 		exchangeRateInterval := int64(k.GetParam(ctx, types.KeyExchangeRateInterval))
-		if epochNumber%exchangeRateInterval == 3 { // allow a few blocks from UpdateUndelegatedBal to avoid conflicts
-			// TODO(TEST-97) update only when balances, delegatedBalances and stAsset supply are results from the same block
-			k.ProcessUpdateDelegatedBalance(ctx)
-		}
-		if epochNumber%exchangeRateInterval == 6 && (epochNumber > 100) { // allow a few blocks from UpdateDelegatedBal to avoid conflicts & wait until chain has registered zones to calc exch rate
-			// TODO(TEST-97) update only when balances, delegatedBalances and stAsset supply are results from the same block
-			k.ProcessUpdateExchangeRate(ctx)
-		}
+		if epochNumber%exchangeRateInterval == 0 && (epochNumber > 100) { // allow a few blocks from UpdateUndelegatedBal to avoid conflicts
+			// GET LATEST HEIGHT
+			// TODO(NOW) wrap this into a function
+			var latestHeightGaia int64 // defaults to 0
+			// get light client's latest height
+			connectionID := "connection-0"
```

Comment:
imo we should loop through all the registered host zones here, and then skip the host zone is delegation account isn't defined yet (which should get rid of the issue where we have to wait 100 epochs)

curious what you think though!

Reply:
I think this is partially resolved by [45e6333](https://github.com/Stride-Labs/stride/pull/47/commits/45e6333c258b2dcfcc0ba273b301d3977ca1c17c), we still need to generalize this to multiple host zones though!